README.md

```diff
@@ -1 +1,29 @@
-Refer to respective folders for assignments
\ No newline at end of file
+# nick-fury
+
+# How to run
+Add all your secret keys in secret.py file\
+\
+There is a file createdatabase.py. Run this file it will ask for password and will create databases.\
+After that. you have to run cronjob.py. Again it will also ask you for password of you mysql. After enterning the password it will run in every 5 minute.\
+\
+There is a file called check_availability.py. This file supposed to be run seperately\
+You have to run it (python check_availability "CommonName"). It will give you the latest information of no. of bikes of this place. Example is below.\
+![](images/screenshot.png)
+
+There is a file scrape.py. This is a file for all the backend operation. You don't need to run that.\
+
+# Approach and assumption.
+a. Two databases for storing real-time bike availability data and bike availability data at different time snapshot. Schema would be same for both databases.\
+b. Two table in each database\
+c. One table is for storing terminal id, common name, terminal name and terminal id is our primary key.
+d. Second table is for storing terminal name, no. of bikes available, no. of empty docks, no. of docks, modified time. For real-time database terminal name would be our primary key and for other database (terminal,modified time) would be our primary key.\
+e. Fetching data from online API in python using request module. Structuring the data.\
+f. For real time database. Deleting rows which is being modified in online API and re uploading the data in database.\
```

Comment:
Is there any reason you are choosing to delete and insert rather than updating?

Reply:
There is no specific reason for that. I selected wrong primary key for time stamp database so "duplicate key on update" query on this database was behaving like real-time database. I figured out my mistake while updating create_database.py file today so i changed my approach and update that. And "duplicate key on update" operation is better as it's improve execution time of job.